### PR TITLE
Atom Tools: factor out lighting and model preset selection into generic asset selection widgets

### PR DESCRIPTION
**Pull request by gadams3**
•	Renamed asset grid dialog to asset selection grid and updated it to monitor the asset catalog instead of passing in an asset list
•	Added utility function for generating a display name from a source file name
•	Remove display name from model and lighting presets which will now use the new display name function
•	Added asset filter callback to combo box constructor
•	Ensure both asset selection widgets are selecting the first item by default
•	Updated viewport settings to use settings registry instead of user settings object
•	Simplified viewport settings request bus to use set, load, save commands and focus on one active preset
•	Simplified viewport settings notification bus to only have one notification for any settings changes
•	Got rid of lighting and model preset pointer and copying settings and presets by value
•	No longer pre-loading all presets on startup but caching them whenever they're loaded or changed
Signed-off-by: Guthrie Adams <guthadam@amazon.com>

### CODE REVIEW

**Review by AMZN-mnaumov — APPROVED** [review]

File: Gems/Atom/Tools/MaterialEditor/Code/Source/Viewport/MaterialViewportWidget.cpp [hunk 1]
```diff
@@ -66,6 +65,12 @@ namespace MaterialEditor
         : AtomToolsFramework::RenderViewportWidget(parent)
         , m_toolId(toolId)
     {
+        // The viewport context created by AtomToolsFramework::RenderViewportWidget has no name.
+        // Systems like frame capturing and post FX expect there to be a context with DefaultViewportContextName
+        auto viewportContextManager = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
+        const AZ::Name defaultContextName = viewportContextManager->GetDefaultViewportContextName();
+        viewportContextManager->RenameViewportContext(GetViewportContext(), defaultContextName);
```
Comment (moraaar) [inline Gems/Atom/Tools/MaterialEditor/Code/Source/Viewport/MaterialViewportWidget.cpp:72]: The viewport context created by `AtomToolsFramework::RenderViewportWidget` does have a name. It's given a unique name in `ViewportContextManager::CreateViewportContext` called from `RenderViewportWidget::InitializeViewportContext`.

Renaming the viewport context to `DefaultViewportContextName` will conflict in the future when there are more than one viewport opened at the same time, either with other Material Editor viewports or the main editor viewport. We don't have to fix this now, but it's important to keep it in mind.

If there are systems that are expecting a specific name for the viewport context, those will need to be improved to work with any viewport name.

**Review by moraaar — APPROVED** [review]
LGTM, just some comments but nothing blocking.

**Review by cgalvan — APPROVED** [review]

File: Gems/Atom/Tools/MaterialEditor/Code/Source/Viewport/MaterialViewportWidget.cpp (hunk 1 above)
Comment (gadams3) [reply Gems/Atom/Tools/MaterialEditor/Code/Source/Viewport/MaterialViewportWidget.cpp:72]: This part of the change is restoring code that existed previously. I only realized it was broken while restoring the smoke test script. There needs to be a condition to test for the default or first viewport for this and the msaa function. Needs to be addressed for anything with multiple viewports. Will discuss with @NicholasVanSickle @dmcdiarmid-ly @rgba16f for a follow up change.

File: Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/AssetSelection/AssetSelectionComboBox.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace AtomToolsFramework
+{
+    class AssetSelectionComboBox
+        : public QComboBox
+        , private AzFramework::AssetCatalogEventBus::Handler
+    {
+        Q_OBJECT
+    public:
+        AssetSelectionComboBox(const AZStd::function<bool(const AZ::Data::AssetInfo&)>& filterCallback, QWidget* parent = 0);
```
[the remark this answers is not in the document]
Comment (gadams3) [reply Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/AssetSelection/AssetSelectionComboBox.h:24]: Sure. I will address in a follow up

**Pull request merged**